source/common/http/filter/ip_tagging_filter.cc

```diff
@@ -9,7 +11,38 @@ IpTaggingFilter::~IpTaggingFilter() {}
 
 void IpTaggingFilter::onDestroy() {}
 
-FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap&, bool) {
+FilterHeadersStatus IpTaggingFilter::decodeHeaders(HeaderMap& headers, bool) {
+  bool is_internal_request =
+      headers.EnvoyInternalRequest() && (headers.EnvoyInternalRequest()->value() == "true");
+
+  if ((is_internal_request && config_->requestType() == FilterRequestType::EXTERNAL) ||
+      (!is_internal_request && config_->requestType() == FilterRequestType::INTERNAL) ||
+      (!config_->runtime().snapshot().featureEnabled("ip_tagging.http_filter_enabled", 100))) {
+    return FilterHeadersStatus::Continue;
+  }
+
+  std::vector<std::string> tags =
+      config_->trie().getTags(callbacks_->requestInfo().downstreamRemoteAddress());
+
+  if (!tags.empty()) {
+    HeaderString& header = headers.insertEnvoyIpTags().value();
+    if (!header.empty()) {
+      header.append(", ", 2);
+    }
+    if (tags.size() > 1) {
```

Comment:
This check seems redundant, since it's within the `if (!tags.empty())` block and nothing prior to this within the block can modify `tags`.

Reply:
This is to differentiate between joining one tag and multiple tags. I'll add a comment.

Reply:
Oops, I hadn't read the following lines with the `else` statement. It makes sense now. No need for a comment.